### ISSUE TITLE
Fix sequence-of-sequence

### ISSUE THREAD

**Issue by andrewdavidmackenzie** [andrewdavidmackenzie/flow#2061]
```
Running unittests examples/sequence-of-sequences/main.rs (target/debug/examples/sequence_of_sequences-a3af467001d40550)

running 1 test
4d3
< 2
6c5,6
< 3
---
> 2
> 2
test test::test_sequence_of_sequences_example ... FAILED

failures:

---- test::test_sequence_of_sequences_example stdout ----

	Running example: flowr/examples/sequence-of-sequences
		Runner: flowrcli
		STDIN is read from test.stdin
		Arguments are read from test.args
		STDOUT is saved in test.stdout
		STDERR is saved in test.stderr
		File output is saved in test.file
	Command line: 'flowrcli --native manifest.json'
thread 'test::test_sequence_of_sequences_example' panicked at flowr/utilities/src/lib.rs:186:9:
Contents of 'flowr/examples/sequence-of-sequences/test.stdout' doesn't match the expected contents in 'flowr/examples/sequence-of-sequences/expected.stdout'
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    test::test_sequence_of_sequences_example

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 1.09s

error: test failed, to rerun pass `-p flowr --example sequence-of-sequences`
Error: The process '/home/runner/.cargo/bin/cargo' failed with exit code 101
```

**Comment by andrewdavidmackenzie** [comment]
```
running 1 test
test test::test_sequence_example ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 1.09s

     Running unittests examples/sequence-of-sequences/main.rs (target/debug/examples/sequence_of_sequences-70eb752fe6da0c57)

running 1 test
10d9
< 5
12c11,12
< 6
---
> 5
> 5
test test::test_sequence_of_sequences_example ... FAILED

failures:

---- test::test_sequence_of_sequences_example stdout ----

	Running example: flowr/examples/sequence-of-sequences
		Runner: flowrcli
		STDIN is read from test.stdin
		Arguments are read from test.args
		STDOUT is saved in test.stdout
		STDERR is saved in test.stderr
		File output is saved in test.file
	Command line: 'flowrcli --native manifest.json'
thread 'test::test_sequence_of_sequences_example' panicked at flowr/utilities/src/lib.rs:186:9:
Contents of 'flowr/examples/sequence-of-sequences/test.stdout' doesn't match the expected contents in 'flowr/examples/sequence-of-sequences/expected.stdout'
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    test::test_sequence_of_sequences_example

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 1.08s
```

**Comment by andrewdavidmackenzie** [comment]
```
Running unittests examples/sequence-of-sequences/main.rs (target/debug/examples/sequence_of_sequences-b699058bbbd6b9a8)

running 1 test
4a5
> 2
7d7
< 0
8a9,10
> 3
> 4
10d11
< 5
12a14,15
> 5
> 4
14c17
< 7
---
> 2
16a20,22
> 5
> 4
> 8
18c24
< 9
---
> 2
test test::test_sequence_of_sequences_example ... FAILED

failures:

---- test::test_sequence_of_sequences_example stdout ----

        Running example: flowr/examples/sequence-of-sequences
                Runner: flowrcli
                STDIN is read from test.stdin
                Arguments are read from test.args
                STDOUT is saved in test.stdout
                STDERR is saved in test.stderr
                File output is saved in test.file
        Command line: 'flowrcli --native manifest.json'
thread 'test::test_sequence_of_sequences_example' panicked at flowr/utilities/src/lib.rs:186:9:
Contents of 'flowr/examples/sequence-of-sequences/test.stdout' doesn't match the expected contents in 'flowr/examples/sequence-of-sequences/expected.stdout'
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by andrewdavidmackenzie** [comment]
This is a design/conceptual problem.

flow initializers (by definition sent from outside a flow) should act like a function sending to the flow, send values one by one, and block (and not send) until the target flow goes idle (consuming loopbacks, internal connection values etc).

When this does not happen, values or initializers coming from outside a flow get inserted into the flow of values in the "inner loop" of the sub-flow, instead of waiting for it to complete (a "loop" let's call it) and go idle, waiting for new input from the later above (parent flow).

Un-ignore the test, or run sequence of sequences manually with debug output and observe the sequence of events to identify when this happens.

Then analyze the code, and design docs, and the expected output to determine what should be the order.

Then propose design or implementation changes that would fix this and the more general case of nested flows like this.

Ask clarifying questions.

**Comment by andrewdavidmackenzie** [comment]
## Root Cause Analysis

Traced execution with debug logging. The issue is a **premature flow idle detection** caused by the interaction between output blocking and flow idle checks.

### Execution trace for the first iteration (limit=1):

1. **Job #2**: `compare_switch #3` (flow_id=2) runs with [limit=1, start=0]
   - Sends `right-lte=0` to stdout (#4, flow_id=0) — creates **cross-flow block** on #3
   - Sends `left-gt=1` loopback to self:0 — fills left input
   - Sends `right-lte=0` to `add #2:0` — fills add's i1
   - Job for `add #2` created, added to busy_flows

2. **Job #5**: `add #2` (flow_id=2) runs with [0, 1] → outputs 1
   - Loopback 1 to self:1 (fills add's i2 for next run)
   - Sends 1 to `compare_switch #3:1` (fills right input)
   - At this point, compare_switch #3 has both inputs [left=1, right=1] — ready to run
   - **BUT**: `create_jobs_or_block` checks `block_exists(#3)` — compare_switch #3 IS blocked on output because stdout (#4) hasn't consumed the value from step 1
   - So compare_switch #3 goes to `blocked` set, NOT `busy_flows`

3. **Idle check**: `unblock_flows` removes `add #2` from `busy_flows`
   - `busy_flows` for flow_id=2 is now empty
   - Flow #2 declared idle → **external senders unblocked** → outer sequence sends next value

### The problem:

compare_switch #3 has all inputs ready and a pending job, but it's in `blocked` (not `busy_flows`) because its output destination (stdout) hasn't consumed yet. The flow appears idle because `busy_flows` doesn't include blocked functions.

### Why simple fixes don't work:

- **Adding blocked functions to the idle check** causes deadlock: the flow never goes idle, so the outer sequence's blocks (which resolve on flow idle) never clear
- **Speculatively adding destinations to busy_flows** causes either deadlocks (entries never removed) or incorrect behavior (entries removed too early)

### The real issue:

The flow idle detection conflates two different concepts:
1. "No functions in this flow are executing" (what `busy_flows` tracks)
2. "This flow has no pending work" (what we actually need)

A function that is ready to run but blocked on cross-flow output IS pending work — it will proceed once the other flow processes its job. But treating it as "busy" creates circular dependencies with the flow idle → unblock mechanism.

### Possible approaches:

1. **Deferred idle check**: Don't check flow idle in `retire_a_job`. Instead, check after ALL ready jobs in the current dispatch cycle have been processed
2. **Priority scheduling**: Process same-flow consumers (stdout) before checking idle, ensuring blocks are resolved
3. **Separate "pending" state**: Track functions that are ready but output-blocked separately from busy_flows, and include them in the idle check only for same-flow blocked destinations

**Comment by andrewdavidmackenzie** [comment]
## Experiment: has_blocked guard breaks arrays example

Tested the `has_blocked` guard in isolation on master (no other changes). It creates a deadlock in the `arrays` example:

- Function #4 (flow=1) is blocked sending to function #0 (flow=0)
- The guard prevents flow #1 from going idle
- But flow #1 going idle is needed to unblock external senders via `flow_blocks`
- Deadlock

The guard can't distinguish between:
1. "Blocked because of nested loop back-pressure" (sequence-of-sequences — needs to wait)
2. "Blocked sending to parent flow" (arrays — will resolve naturally, must not prevent idle)

Both cases have a function in `self.blocked` within the sub-flow. The difference is the DIRECTION of the block relative to the flow hierarchy.

Need a different approach that doesn't use the blocked set for idle detection.
